### PR TITLE
Fix Chart Settings

### PR DESCRIPTION
**Pull request by chakrabortyr**
## Description
Removes spurious JSON.encode/decode calls for chart settings in the Usage/Metric Explorer Tabs

## Motivation and Context
Already perfectly valid JSON, encoding/decoding is redundant. 
 
## Tests performed

Ran the tests. Also spun up a new Docker and verified that I could still view/modify a saved chart before and after this change.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project as found in the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

File: html/gui/js/modules/Usage.js [hunk 1]
```diff
@@ -1454,7 +1454,7 @@ Ext.extend(XDMoD.Module.Usage, XDMoD.PortalModule, {
                             chartContainer.setHeight(CCR.xdmod.ui.thumbHeight * chartThumbScale);
                             chartContainer.mask('Loading...');
 
-                            var chart_params = Ext.apply({}, Ext.util.JSON.decode(r.get('chart_settings')), menuParams);
+                            var chart_params = Ext.apply({}, r.get('chart_settings'), menuParams);
```
Comment (jpwhite4) [inline html/gui/js/modules/Usage.js]: What is the impact of this change on any saved charts in the report generator?

Comment (chakrabortyr) [reply html/gui/js/modules/Usage.js]: The chart settings appear to be saved in moddb as RESTful query parameters, which are then encoded as JSON and sent to the frontend. So we should be okay.

Comment (jpwhite4) [reply html/gui/js/modules/Usage.js]: Also need to check the app kernels module is not impacted by this change.

**Comment by jpwhite4** [comment]
Don't forget that you have to update all the places in the frontend code that processes the chart setting data. This includes but is not limited to:
- The `reloadChartFunc` in Usage.js
- The `fromJSON` function in ChartToolbar.js
- The `getChartParameters` function in Usage.js
- The chart store load callback in Usage.js

File: html/gui/js/modules/Usage.js [hunk 2]
```diff
@@ -1925,11 +1925,11 @@ Ext.extend(XDMoD.Module.Usage, XDMoD.PortalModule, {
                         if (n.attributes.supportsAggregate === false) {
                             // do not overide chart settings for nodes that correspond to statistics that must be
                             // shown in timeseries mode (note for future: there should be no statistics like this in XDMoD).
-                            var decoded = JSON.parse(n.attributes.chartSettings);
+                            var decoded = n.attributes.chartSettings;
```
Comment (jpwhite4) [inline html/gui/js/modules/Usage.js]: Hmm. Javascript objects are assigned by reference whereas the original code assigned by value. In the original code the value of n.attributes.chartSettings was not modified. However it is now bein modified in this code. What is the impact of this change?

File: html/gui/js/modules/Usage.js [hunk 3]
```diff
@@ -2659,12 +2659,14 @@ Ext.extend(XDMoD.Module.Usage, XDMoD.PortalModule, {
                         'Chart: ' + chartStore.getAt(0).get('title') + ', Params: ' + chartStore.getAt(0).get('params_title')
                     );
 
-                    var serverChartSettings = chartStore.getAt(0).get('chart_settings').replace(/`/g, '"');
-                    if (serverChartSettings === '') {
-                        serverChartSettings = n.attributes.defaultChartSettings;
+                    var serverChartSettings = Ext.apply({}, chartStore.getAt(0).get('chart_settings'));
+
+                    if (!serverChartSettings) {
+                        serverChartSettings = Ext.apply({}, n.attributes.defaultChartSettings);
```
Comment (plessbd) [inline html/gui/js/modules/Usage.js]: why did this change from pass by reference to pass by value?

File: .gitignore [hunk 4]
```diff
@@ -4,6 +4,7 @@
 *.orig
 *.log
 *.rej
+*.sarif
```
Comment (plessbd) [inline .gitignore]: this is not related to this change please remove it.

File: html/gui/js/modules/Usage.js [hunk 5]
```diff
@@ -1083,7 +1083,7 @@ Ext.extend(XDMoD.Module.Usage, XDMoD.PortalModule, {
             var newStartDate = self.getDurationSelector().getStartDate().format('Y-m-d');
             var newEndDate = self.getDurationSelector().getEndDate().format('Y-m-d');
             var aggregationUnit = self.getDurationSelector().getAggregationUnit();
-            var chartSettings = JSON.decode ? JSON.decode(n.attributes.chartSettings) : JSON.parse ? JSON.parse(n.attributes.chartSettings): {};
+            var chartSettings = Ext.apply({}, n.attributes.chartSettings);
```
Comment (jpwhite4) [inline html/gui/js/modules/Usage.js]: I thought that Ext.apply was a shallow copy. Is this safe to do here?

File: html/gui/js/modules/Usage.js [hunk 6]
```diff
@@ -1925,11 +1925,11 @@ Ext.extend(XDMoD.Module.Usage, XDMoD.PortalModule, {
                         if (n.attributes.supportsAggregate === false) {
                             // do not overide chart settings for nodes that correspond to statistics that must be
                             // shown in timeseries mode (note for future: there should be no statistics like this in XDMoD).
-                            var decoded = JSON.parse(n.attributes.chartSettings);
+                            var decoded = Ext.apply({}, n.attributes.chartSettings);
```
Comment (jpwhite4) [inline html/gui/js/modules/Usage.js]: There is no need to create an object copy here since this code is supposed to modify the node itself.
```js
if (n.attributes.supportsAggregate === false) {
    n.attributes.dataset_type = 'timeseries';
    n.attributes.display_type = 'line';
    n.attributes.swap_xy = false;
}
```

File: html/gui/js/modules/Usage.js [hunk 7]
```diff
@@ -2659,12 +2659,14 @@ Ext.extend(XDMoD.Module.Usage, XDMoD.PortalModule, {
                         'Chart: ' + chartStore.getAt(0).get('title') + ', Params: ' + chartStore.getAt(0).get('params_title')
                     );
 
-                    var serverChartSettings = chartStore.getAt(0).get('chart_settings').replace(/`/g, '"');
-                    if (serverChartSettings === '') {
-                        serverChartSettings = n.attributes.defaultChartSettings;
+                    var serverChartSettings = Ext.apply({}, chartStore.getAt(0).get('chart_settings'));
+
+                    if (!serverChartSettings) {
```
Comment (jpwhite4) [inline html/gui/js/modules/Usage.js]: This won't work

File: html/gui/js/modules/Usage.js (hunk 5 above)
Comment (chakrabortyr) [reply html/gui/js/modules/Usage.js]: Yes, we don't have nested objects in the chart settings.

**Comment by jpwhite4** [comment]
The controllers are planned to go away in 9.5 so there is no point in changing them now.

**Comment by chakrabortyr** [comment]
How anticlimactic
